internal/providers/terraform/aws/eks_node_group.go

```diff
@@ -0,0 +1,58 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetNewEKSNodeGroupItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_eks_node_group",
+		RFunc: NewEKSNodeGroup,
+	}
+}
+
+func NewEKSNodeGroup(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+	costComponents := make([]*schema.CostComponent, 0)
+	costComponents = append(costComponents, eksComputeCostComponent(d))
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func eksComputeCostComponent(d *schema.ResourceData) *schema.CostComponent {
+	region := d.Get("region").String()
+	scalingConfig := d.Get("scaling_config").Array()[0]
+	desiredSize := int(scalingConfig.Get("desired_size").Int())
+	instanceType := "t3.medium"
+	if d.Get("instance_type").Exists() {
+		instanceType = d.Get("instance_type").String()
+	}
+	purchaseOptionLabel := "on-demand"
+
+	return &schema.CostComponent{
+		Name:           fmt.Sprintf("Linux/UNIX usage (%s, %s)", purchaseOptionLabel, instanceType),
+		Unit:           "hours",
+		UnitMultiplier: desiredSize,
+		HourlyQuantity: decimalPtr(decimal.NewFromInt(1)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("aws"),
+			Region:        strPtr(region),
+			Service:       strPtr("AmazonEC2"),
+			ProductFamily: strPtr("Compute Instance"),
+			AttributeFilters: []*schema.AttributeFilter{
```

Comment:
Seems like we should also set `tenancy` as these filters match multiple tenancy options, see the `instance.go` for an example. It doesn't look like users can change that in `aws_eks_node_group` so it could be hardcoded to `Shared`.